### ISSUE TITLE
wiki test & making sure all ok

### ISSUE THREAD

**Issue by despiegk** [threefoldtecharchive/jumpscaleX_core#64]
- [x] wiki_test package in threebot repo which 
    - [x] has example wiki pages to demonstrate all below in the package
    - [x] load content using myjobs 
    - [x] make sure threebot properly loaded with all features
    - [x] if started with public ip address available make sure nginx/letsencrypt work
- [x] gslide macro works (get .png from a slide in gslides presentation) dynamically
- [x] new macro to get markdown content from gdoc and put in local doc to further process like any markdown doc (in other words allows us to collaborate on gdoc for markdown doc but yet still have all other features working)
- [x] tests with include macro (many scenarios)
- [x] tests with link feature (many scenarios)
- [x] test the search well (maybe some test script which is testing against sonic)
- [x] deploy primary dns server on a DO machine to leave there for tests
- [x] make sure we have a client access to it (actor based client) to configure this DNS, can be for everyone today using jumpscale because of preconfigered secret to use in actor

### test to do

- create DNS hoster machine on DO
- point subdomain web.grid.tf to point to this DNS digital ocean machine (no name.com)
    - in other words all DNS requests for xxx.web.grid.tf endup on this machine
- configure client access to this DNS server through an actor (new package probably, ok to work with predefined secret hardcoded for now), so we can use to configure DNS
- create TEST machine on digital ocean 
- deploy the testwiki package above on the TEST machine
- use the DNS client to configure e.g. wiki.web.grid.tf
- now check with script the wiki works http(s) and content comes from it

**Comment by waleedhammam** [comment]
deploy script; 
2 machines; 1 for dns and routing; the other for deploying wikis.
https://gist.github.com/waleedhammam/08149e0a2db723735d893482da1de1ad
still working on ssl and certificates.
site is live at: http://wiki.web.grid.tf/

**Comment by despiegk** [comment]
see also https://github.com/threefoldtech/jumpscaleX_threebot/issues/10

**Comment by waleedhammam** [comment]
test wikis macros:
wikis package should be running `kosmos -p 'j.threebot.package.wikis.start()'`
- create a file for example macros.py:
echo in it the following
```python
def load_wiki(name, url):
	wiki = j.tools.markdowndocs.load(path=url, name=name)
	wiki.write()
```
add the test wikis
```python
import macros
j.servers.myjobs.schedule(macros.load_wiki, "testmacros","https://github.com/Dinaamagdy/test_custom_md/tree/master/docs" )
```
then: http://sitename/wiki/testmacros to test it

**Comment by serboctor** [comment]
Macro to get markdown from gdoc https://github.com/threefoldtech/jumpscaleX_libs/blob/development/JumpscaleLibs/tools/markdowndocs/macros/docs/markdown.md

**Comment by waleedhammam** [comment]
ssl fixes: https://github.com/threefoldtech/jumpscaleX_core/commit/b01bef6a4d09e8a50a49d8f6e24c7f1744d46bb7 https://github.com/threefoldtech/jumpscaleX_core/commit/5015c89be28c8dd9e9d74477fd672a03716c3bbb  https://github.com/threefoldtech/jumpscaleX_core/commit/c2a6a4248dd6a8f3808d633665e18ecf0bb0ed87
ssl working now at: https://wikis.web.grid.tf/

updated deployment script: https://gist.github.com/waleedhammam/08149e0a2db723735d893482da1de1ad

will do more tweaks to the script.
write more details and documents

DNS Machine: `167.71.11.217`
Wikis Machine: `157.245.72.202`

**Comment by waleedhammam** [comment]
Intergrated the deploy tool to jumpscale at: https://github.com/threefoldtech/jumpscaleX_libs_extra/tree/development/JumpscaleLibsExtra/tools/threebot_deploy

currently adding readme to it and more tests

**Comment by Pishoy** [comment]
verified 
according to docs https://github.com/threefoldtech/jumpscaleX_libs_extra/tree/development_stabilize/JumpscaleLibsExtra/tools/threebot_deploy

## Deploy DNS Hoster
```
dns_machine = j.tools.threebot_deploy.get("testdns", do_machine_name="testgateway", do_token="3faa24", do_project_name="codescalers", ssh_key="Bishoy")
dns_machine.create_new_do_machine()
gitdns_machine.jsx_install(branch="development")
dns_machine.install_tcprouter_coredns()
```
- Setup mytestdns.grid.tf as a ns record which is redirect to A record to the machine that hosting the DNS 
dns_machine.add_dns_record(subdomain="wikis", domain="mytestdns.grid.tf", wikis_machine_ip="167.71.68.80", wikis_machine_port="443") 
 - you can login to dns host and check redis 
![image](https://user-images.githubusercontent.com/21987477/64924180-f20c0c00-d7e1-11e9-8d71-06bd7b359121.png)


## Deploy 3bot wikis
```
wikis_machine = j.tools.threebot_deploy.get("testwikis", do_machine_name="testwikis", do_token="8a677bbd", do_project_name="codescalers", ssh_key="Bishoy")
wikis_machine.create_new_do_machine() 
wikis_machine.jsx_install(branch="development")
wikis_machine.deploy_wikis()
```
- now we can access wiki by link https://wikis.mytestdns.grid.tf/wiki/foundation#/
